include/boost/numeric/ublas/tensor/static_tensor.hpp

```diff
@@ -0,0 +1,271 @@
+        using difference_type 			= typename array_type::difference_type;
+        using value_type      			= typename array_type::value_type;
+
+        using reference       			= typename array_type::reference;
+        using const_reference 			= typename array_type::const_reference;
+
+        using pointer         			= typename array_type::pointer;
+        using const_pointer   			= typename array_type::const_pointer;
+
+        using iterator        			= typename array_type::iterator;
+        using const_iterator  			= typename array_type::const_iterator;
+
+        using reverse_iterator        	= typename array_type::reverse_iterator;
+        using const_reverse_iterator  	= typename array_type::const_reverse_iterator;
+
+        using tensor_temporary_type     = self_type;
+        using storage_category          = dense_tag;
+
+        using extents_type              = typename detail::tensor_traits<self_type>::extents_type;
+        using strides_type              = strides_t<extents_type,layout_type>;
+
+        using matrix_type               = typename super_type::matrix_type;
+        using vector_type               = typename super_type::vector_type;
+
+        constexpr static_tensor() = default;
+
+        constexpr static_tensor( array_type const& a ){
+            std::copy(a.begin(), a.end(), super_type::begin());
+        }
+
+        constexpr static_tensor( value_type const& i ){
+            std::fill(super_type::begin(), super_type::end(), i);
+        }
+
+        constexpr static_tensor( matrix_type const& v )
+        {
+            static_assert(extents_type::_size == 2
+                , "Error in boost::numeric::ublas::tensor(const matrix_type &v)"
+                                        " : invalid extents size");
+            
+            if( !( super_type::size(0) == v.size1() || super_type::size(1) == v.size2() ) ){
+                throw std::runtime_error("Error in boost::numeric::ublas::tensor(const matrix_type &v)"
+                                        " : rank of extents are not correct, please check!");
+            }
+            std::copy(v.data().begin(), v.data().end(),super_type::data_.begin());
+        }
+
+        constexpr static_tensor( matrix_type && v )
+        {
+            static_assert(extents_type::_size == 2
+                , "Error in boost::numeric::ublas::tensor(const matrix_type &v)"
+                                        " : invalid extents size");
+            
+            auto const sz = v.size1() * v.size2();
+            
+            if( !( super_type::size(0) == v.size1() || super_type::size(1) == v.size2() ) ){
+                throw std::runtime_error("Error in boost::numeric::ublas::tensor(matrix_type &&v)"
+                                        " : rank of extents are not correct, please check!");
+            }
+
+            for(auto i = size_type{}; i < sz; ++i){
+                super_type::data_[i] = std::move(v.data()[i]);
+            }
```

Comment:
Instead of a `for` loop, I recommend to use `std::move` algorithm.

Reply:
done.